tempodb/encoding/blocker.go

```diff
@@ -0,0 +1,51 @@
+package encoding
+
+import (
+	"context"
+
+	"github.com/grafana/tempo/pkg/tempopb"
+	"github.com/grafana/tempo/tempodb/encoding/common"
+)
+
+type Finder interface {
+	FindTraceByID(ctx context.Context, id common.ID) (*tempopb.Trace, error)
+}
+
+type Searcher interface {
+	Search(ctx context.Context, req *tempopb.SearchRequest, opts ...SearchOption) (*tempopb.SearchResponse, error)
+}
```

Comment:
Nice additions. Curious if these could go into a `BackendBlock` interface, that would be really neat

```
type BackendBlock interface {
    Finder
    Searcher
    Iterator(?)
}
```

But unsure if we can achieve that level of abstraction outside of tempodb

Reply:
I did originally start with a single interface for all block reads, but realized that Tempo never performs them at the same time.  It's either trace lookup or a search.  Therefore the split interface makes sense, and should be improve readability and testability.